faiss/IndexIVFFlat.cpp

```diff
@@ -166,6 +192,26 @@ struct IVFFlatScanner : InvertedListScanner {
         return nup;
     }
 
+    size_t scan_codes_batched(
+            size_t list_size,
+            const uint8_t* codes,
+            const idx_t* ids,
+            float* simi,
+            idx_t* idxi,
+            size_t k) const override {
+        const float* list_vecs = (const float*)codes;
+        if (metric == METRIC_INNER_PRODUCT) {
+            float_minheap_array_t res = {nx, size_t(k), idxi, simi};
+            knn_inner_product(xi, list_vecs, d, nx, list_size, &res);
+            update_res_ids<float_minheap_array_t>(&res, nx * k, ids);
```

Comment:
This should handle -1 entries. 
Please specify that the input heap should be empty (ie. not contain previous results to combine with).

Reply:
handled by the last commit.